next/src/components/ui/SectionHeading/SectionHeading.tsx

```diff
@@ -0,0 +1,14 @@
+import Markdown from '../markdown';
+import styles from './SectionHeading.module.scss';
+import type { SectionHeadingTypes } from './SectionHeading.types';
+
+export default function SectionHeading({ heading, subheading, index }: SectionHeadingTypes) {
+  const Heading = index === 0 ? 'h1' : 'h2';
+
+  return (
+    <Heading className={styles.heading}>
+      <Markdown.span className={styles.main}>{heading}</Markdown.span>
+      {Boolean(subheading) && <Markdown.span className={styles.sub}>{subheading!}</Markdown.span>}
+    </Heading>
+  );
```

Comment:
- Basing `Heading` level on index can be good idea for a components (because of the order of the sections), but I would rather use there a `h1`, `h2`, `h3`, `h4`, `h5`, `h6` type props, instead of the `index`, because now at a glance I doesn't know how to change the heading level.
- I would recommend you simplify the naming for just `Heading` not `SectionHeading`
- Instead of `Boolean(subheading)` I would rather do just `!!subheading`. [Double negation operator in JS](https://dev.to/ypdev19/learning-about-the-operator-in-javascript-2c0o)
- Why are you using the `subheading!` while passing the value to `Markdown`, that's not a good idea at a glance

Reply:
- It will be more dynamic solution for sure, ill change that.
- Got it.
- I had to use `subheading!` beacouse the Boolean(x) does not work well with Typescript. I've now changed to !!x, and it's working well.